lib/chef/provider/ifconfig.rb

```diff
@@ -57,30 +57,78 @@ def load_current_resource
         @ifconfig_success = true
         @interfaces = {}
 
-        @status = shell_out("ifconfig")
-        @status.stdout.each_line do |line|
-          if !line[0..9].strip.empty?
-            @int_name = line[0..9].strip
-            @interfaces[@int_name] = { "hwaddr" => (line =~ /(HWaddr)/ ? ($') : "nil").strip.chomp }
-          else
-            @interfaces[@int_name]["inet_addr"] = (line =~ /inet addr:(\S+)/ ? Regexp.last_match(1) : "nil") if line =~ /inet addr:/
-            @interfaces[@int_name]["bcast"] = (line =~ /Bcast:(\S+)/ ? Regexp.last_match(1) : "nil") if line =~ /Bcast:/
-            @interfaces[@int_name]["mask"] = (line =~ /Mask:(\S+)/ ? Regexp.last_match(1) : "nil") if line =~ /Mask:/
-            @interfaces[@int_name]["mtu"] = (line =~ /MTU:(\S+)/ ? Regexp.last_match(1) : "nil") if line =~ /MTU:/
-            @interfaces[@int_name]["metric"] = (line =~ /Metric:(\S+)/ ? Regexp.last_match(1) : "nil") if line =~ /Metric:/
+        @ifconfig_version = nil
+
+        @net_tools_version = shell_out("ifconfig --version")
+        @net_tools_version.stderr.each_line do |line|
+          if line =~ /^net-tools (\d+.\d+)/
+            @ifconfig_version = line.match(/^net-tools (\d+.\d+)/)[1]
+          end
+        end
+
+        if @ifconfig_version.nil?
+          raise "net-tools not found - this is required for ifconfig"
+        elsif @ifconfig_version.to_f < 2.0
+          @status = shell_out("ifconfig")
+          @status.stdout.each_line do |line|
```

Comment:
It would be really good to get some example output here and on line 97 so that we know what this should look like and what the expectations are.